packages/react/src/components/Forms/Select/mono.stories.tsx

```diff
@@ -134,6 +139,57 @@ export const SearchableUncontrolled = {
   },
 };
 
+export const AsyncSearchableUncontrolled = {
+  render: Template,
+
+  args: {
+    label: "Select a city",
+    options: async (context: ContextOptionAsaCallback) =>
+      new Promise((resolve) => {
+        const arrayCities = [
+          {
+            label: "Paris",
+            value: "paris",
+          },
+          {
+            label: "Panama",
+            value: "panama",
+          },
+          {
+            label: "London",
+            value: "london",
+          },
+          {
+            label: "New York",
+            value: "new_york",
+          },
+          {
+            label: "Tokyo",
+            value: "tokyo",
+          },
+        ];
+
+        // simulate a delayed response
+        setTimeout(() => {
+          const stringSearch = context?.search ?? undefined;
+
+          const filterOptions = (arrayOptions: Option[], search: string) =>
+            arrayOptions.filter((option) =>
+              option.label.toLocaleLowerCase().includes(search.toLowerCase()),
+            );
+
+          const arrayOptions: Option[] = stringSearch
+            ? filterOptions(arrayCities, stringSearch)
+            : arrayCities;
+
+          resolve(arrayOptions);
+        }, 500);
+      }),
+    defaultValue: OPTIONS[4].value,
```

Comment:
The default value does not seem to be working with the async mode:

<img width="470" alt="Capture d’écran 2024-03-19 à 11 01 13" src="https://github.com/openfun/cunningham/assets/9628870/c9f6ea73-c32c-41c1-bb64-5cf7b6c5bc0e">